biz.aQute.launcher/src/aQute/launcher/Launcher.java

```diff
@@ -1199,16 +1205,39 @@ public void report(PrintStream out) {
 					out.println("Id    Levl State Modified      Location");
 
 					for (int i = 0; i < bundles.length; i++) {
+						String lastModified = "<>";
 						String loc = bundles[i].getLocation();
-						loc = loc.replaceAll("\\w+:", "");
-						File f = new File(loc);
+						try {
+							URI locURI = new URI(loc);
+							int count = 0;
+							WHILE: while (locURI.getScheme() != null) {
+								switch (locURI.getScheme()
+									.toLowerCase()) {
+									case "file" :
+										Path p = Paths.get(locURI);
+										long time = Files.getLastModifiedTime(p)
+											.toMillis();
+										lastModified = FORMAT.format(Files.getLastModifiedTime(p)
+											.toInstant());
+										loc = p.toString();
+										break WHILE;
+									case "reference" :
+										locURI = new URI(locURI.getRawSchemeSpecificPart());
+										break;
+									default :
+										break WHILE;
+								}
+								count++;
```

Comment:
I do not see how recursion could be possible so no use to check some limit. If you insist, make the length of the location. Dont think this is necessary though

Reply:
I think I was just being overly paranoid. I'll remove it.

Reply:
Done.